rust/transaction-filter/src/filter_operator.rs

```diff
@@ -0,0 +1,336 @@
+// use crate::traits::Filterable;
+use crate::{
+    filters::{
+        EventFilter, TransactionRootFilter, UserTransactionRequestFilter, WriteSetChangeFilter,
+    },
+    traits::Filterable,
+};
+use anyhow::Error;
+use aptos_protos::transaction::v1::{transaction::TxnData, Transaction};
+use serde::{Deserialize, Serialize};
+use std::fmt::Debug;
+
+/// These are filters we would expect to be exposed via API
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
+#[serde(deny_unknown_fields)]
+#[serde(tag = "type")]
+pub enum APIFilter {
+    TransactionRootFilter(TransactionRootFilter),
+    UserTransactionRequestFilter(UserTransactionRequestFilter),
+    EventFilter(EventFilter),
+    WriteSetChangeFilter(WriteSetChangeFilter),
+}
+
+impl Filterable<Transaction> for APIFilter {
+    fn is_valid(&self) -> Result<(), Error> {
+        match self {
+            APIFilter::TransactionRootFilter(filter) => filter.is_valid(),
+            APIFilter::UserTransactionRequestFilter(filter) => filter.is_valid(),
+            APIFilter::EventFilter(filter) => filter.is_valid(),
+            APIFilter::WriteSetChangeFilter(filter) => filter.is_valid(),
+        }
+    }
+
+    fn is_allowed(&self, txn: &Transaction) -> bool {
```

Comment:
I don't wanna nerd snipe you but it'd be neat if this function at the top level told you which filter allowed it if this is true 😛

Reply:
what do you mean?

Reply:
Like if a transaction got filtered out, it'd be cool to see which filter exactly it failed on.

Reply:
sigh :-p

this is not technically hard, just a pain in the ass hahaha... Follow up PR?